shopinvader_schema_sale/schemas/sale_order.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+from typing import List
+
+from odoo.addons.extendable_fastapi import StrictExtendableBaseModel
+
+from .amount import SaleAmount
+from .invoicing import InvoicingInfo
+from .sale_order_line import SaleOrderLine
+from .shipping import ShippingInfo
+
+
+class BaseSale(StrictExtendableBaseModel):
+    id: int
+    state: str
+    name: str
+    client_order_ref: str | None = None
+    date_order: datetime
+    lines: List[SaleOrderLine]
+    amount: SaleAmount | None = None
+    # TODO discuss about this (should we keep the same schema for this field)
+    shipping: ShippingInfo | None = None
```

Comment:
These are not the same schema. This approach allows to add specific information when depending of the address usage event if the 2 models share the same base information.

Reply:
Sorry, what I mean here is that I have changed the schema.

### Before my change it was

```
    delivery: ShippingAddress | None = None
    invoicing: BillingAddress | None = None
```
see : here https://github.com/shopinvader/odoo-shopinvader/pull/1421/files#diff-58ebe71b902dc742a85164f8e89fb500c247c88eb0aca39d334e565d08923377L35

So the response look like 

```
{
    "delivery": {"name": XXX, "street": YYY...}
    "invoicing": {"name": XXX, "street": YYY...}
}

```





### With my change it's more like in the V1

```
    shipping: ShippingInfo | None = None
    invoicing: InvoicingInfo | None = None
```


So the response look like 

```
{
    "shipping": {"address": {"name": XXX, "street": YYY...}}
    "invoicing": {"address": {"name": XXX, "street": YYY...}}
}

```

The idea is to be able to extend the ShippingInfo or the InvoicingInfo to add extra information specific to the invoicing or shipping information


@thibaultrey what do you prefers ?

Reply:
@sebastienbeau It's ok for me (it's as is in my sale_cart_restapi implementation in v10 :smirk: ) Expect I would like to keep 'delivery' in place of 'shipping'

Reply:
ok for me